### ISSUE TITLE
"indent" rule false positive on "while" statement

### ISSUE THREAD

**Issue by feross** [eslint/eslint#3000]
Using the code on `master` (as of https://github.com/eslint/eslint/commit/1c70834295b064021e6aa6698473375063bffbae), I get a false positive with the `indent` rule enabled.

Code:

```js
while (1 < 2) console.log('hi')
```

Rule:

```
"indent": [2, 2]
```

Incorrect error:

```
t.js:1:15: Expected indentation of 2 characters.
```

**Comment by feross** [comment]
And here's another instance where "indent" gives a false positive:

```js
function salutation () {
  switch (1) {
    case 0: return console.log('hi')
    case 1: return console.log('hey')
  }
}
```

```
t.js:3:13: Expected indentation of 6 characters.
t.js:4:13: Expected indentation of 6 characters.
```

**Comment by gyandeeps** [comment]
Can you file a separate issue for switch case.

For the while `case` discussed above, i am not able to produce the error.

**Comment by feross** [comment]
Sure, separate issue filed: https://github.com/eslint/eslint/issues/3001

**Comment by feross** [comment]
@gyandeeps I just reconfirmed that the issue indeed exists. Are you using the version of eslint that's on `master`?

**Comment by feross** [comment]
Here's a demo: https://github.com/feross/eslint-issue-3000

```
git clone git@github.com:feross/eslint-issue-3000.git
cd eslint-issue-3000
npm install
npm test
```

**Comment by feross** [comment]
Your test case here (https://github.com/eslint/eslint/commit/c46b7a9144bea1241fd86b78a3d72abf1492ae50#diff-a28ba1a60512ed1edcdfca32d70ce897R57) added a `;` to the end, which might be causing the test to pass.

**Comment by gyandeeps** [comment]
I tried it by removing it, still passes.

**Comment by feross** [comment]
@gyandeeps What about my demo repo?

**Comment by ilyavolodin** [comment]
I can confirm that demo is failing (as is, without latest fix)

**Comment by gyandeeps** [comment]
I confirmed the bug using the repo now and it is also because of the semi at the end.
But if I use the same code inside unit test, it passes. :scream:

**Comment by ilyavolodin** [comment]
Hmm... Ok, now that's very interesting. Unit tests shouldn't be doing anything that would cause such behavior...

**Comment by gyandeeps** [comment]
Can anyone else try it in unit test to confirm please?

use this:

```js
{
            code:
            "while (1 < 2) console.log('hi')",
            options: [2]
        }
```

**Comment by ilyavolodin** [comment]
I just verified it locally, and you are correct, it's passing unittests. No idea why, but we really should investigate this.

**Comment by gyandeeps** [comment]
I know why now..... ready everyone *drum roll*
AS i mentioned this issue https://github.com/eslint/espree/issues/164 where if a stement doesn't have a `;` then it make the start and end for location to be not on same line.

Now if you closely look into @feross repo and the js file in it. it has an empty line. that hwy.
Our unit test doesnt have that.

**Comment by ilyavolodin** [comment]
Ok, that makes sense. I got scared there for a second that our unittests where not testing things correctly.

**Comment by lo1tuma** [comment]
working on this
